libs/beacon/src/beacon_manager.cpp

```diff
@@ -297,37 +344,48 @@ BeaconManager::MuddleAddress BeaconManager::VerifyQualComplaint(MuddleAddress co
   PublicKey  rhs;
   PrivateKey s;
   PrivateKey sprime;
-  s.setStr(answer.second.first);
-  sprime.setStr(answer.second.second);
+  bool       s_set{false};
+  bool       sprime_set{false};
+  s.setStr(&s_set, answer.second.first.data());
+  sprime.setStr(&sprime_set, answer.second.second.data());
   // check equation (4)
-  lhs = crypto::mcl::ComputeLHS(group_g_, group_h_, s, sprime);
-  rhs = crypto::mcl::ComputeRHS(from_index, C_ik[victim_index]);
-  if (lhs != rhs)
-  {
-    FETCH_LOG_WARN(LOGGING_NAME, "Node ", cabinet_index_,
-                   " received shares failing initial coefficients verification from node ",
-                   from_index, " for node ", victim_index);
-    return from;
-  }
-  else
+  if (s_set && sprime_set)
   {
-    // check equation (5)
-    bn::G2::mul(lhs, group_g_, s);  // G^s
-    rhs = crypto::mcl::ComputeRHS(from_index, A_ik[victim_index]);
+    lhs = crypto::mcl::ComputeLHS(group_g_, group_h_, s, sprime);
+    rhs = crypto::mcl::ComputeRHS(from_index, C_ik[victim_index]);
     if (lhs != rhs)
     {
       FETCH_LOG_WARN(LOGGING_NAME, "Node ", cabinet_index_,
-                     " received shares failing qual coefficients verification from node ",
+                     " received shares failing initial coefficients verification from node ",
                      from_index, " for node ", victim_index);
-      return answer.first;
+      return from;
     }
     else
     {
-      FETCH_LOG_WARN(LOGGING_NAME, "Node ", cabinet_index_, " received incorrect complaint from ",
-                     from_index);
-      return from;
+      // check equation (5)
```

Comment:
What do these numbers refer to?

Reply:
They refer to equations in the DKG paper. I didn't realise I put that in, I will remove it.